packages/flutter_tools/lib/src/asset.dart

```diff
@@ -203,6 +231,25 @@ class _Asset {
 
   @override
   String toString() => 'asset: $assetEntry';
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(other, this))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final _Asset otherAsset = other;
+    return base == otherAsset.base
+        && assetEntry == otherAsset.assetEntry
+        && relativePath == otherAsset.relativePath
+        && source == otherAsset.source;
+  }
+
+  @override
+  int get hashCode => base.hashCode ^
```

Comment:
Use of `=>` with multi-line body should be avoided, e.g. by replacing with
```dart
@override
int get hashCode => base.hashCode ^ assetEntry.hashCode ^ relativePath.hashCode ^ source.hashCode;
```
or
```dart
@override
int get hashCode {
  return base.hashCode
       ^ assetEntry.hashCode 
       ^ relativePath.hashCode 
       ^ source.hashCode;
}
```